tachyon/zk/plonk/examples/fibonacci/fibonacci1_circuit.h

```diff
@@ -0,0 +1,173 @@
+      return constraints;
+    });
+
+    return Fibonacci1Config<F>(std::move(advice), std::move(selector),
+                               std::move(instance));
+  }
+
+  AssignedRow<F> AssignFirstRow(Layouter<F>* layouter) const {
+    AssignedRow<F> ret;
+    layouter->AssignRegion("first row", [this, &ret](Region<F>& region) {
+      config_.selector.Enable(region, 0);
+
+      AssignedCell<F> a_cell = region.AssignAdviceFromInstance(
+          "f(0)", config_.instance, 0, config_.advice[0], 0);
+      AssignedCell<F> b_cell = region.AssignAdviceFromInstance(
+          "f(1)", config_.instance, 1, config_.advice[1], 0);
+      AssignedCell<F> c_cell = region.AssignAdvice(
+          "a + b", config_.advice[2], 0,
+          [&a_cell, &b_cell]() { return a_cell.value() + b_cell.value(); });
+
+      ret = AssignedRow<F>(std::move(a_cell), std::move(b_cell),
+                           std::move(c_cell));
+    });
+    return ret;
+  }
+
+  AssignedCell<F> AssignRow(Layouter<F>* layouter,
+                            const AssignedCell<F>& prev_b,
+                            const AssignedCell<F>& prev_c) const {
+    AssignedCell<F> ret;
+    layouter->AssignRegion(
+        "next row", [this, &ret, &prev_b, &prev_c](Region<F>& region) {
+          config_.selector.Enable(region, 0);
+
+          // Copy the value from b & c in previous row to a & b in current row
+          const AssignedCell<F> a_cell =
+              prev_b.CopyAdvice("a", region, config_.advice[0], 0);
+          const AssignedCell<F> b_cell =
+              prev_c.CopyAdvice("b", region, config_.advice[1], 0);
+
+          const AssignedCell<F> c_cell = region.AssignAdvice(
+              "a + b", config_.advice[2], 0,
+              [&a_cell, &b_cell]() { return a_cell.value() + b_cell.value(); });
+
+          ret = std::move(c_cell);
+        });
+    return ret;
+  }
+
+  void ExposePublic(Layouter<F>* layouter, const AssignedCell<F>& cell,
+                    RowIndex row) const {
+    layouter->ConstrainInstance(cell.cell(), config_.instance, row);
+  }
+
+ private:
+  Fibonacci1Config<F> config_;
+};
+
+template <typename F, template <typename> class _FloorPlanner>
+class Fibonacci1Circuit : public Circuit<Fibonacci1Config<F>> {
+ public:
+  using FloorPlanner = _FloorPlanner<Fibonacci1Circuit<F, _FloorPlanner>>;
+
+  std::unique_ptr<Circuit<Fibonacci1Config<F>>> WithoutWitness()
+      const override {
+    return std::make_unique<Fibonacci1Circuit>();
+  }
+
+  static Fibonacci1Config<F> Configure(ConstraintSystem<F>& meta) {
+    return Fibonacci1Chip<F>::Configure(meta);
+  }
+
+  void Synthesize(Fibonacci1Config<F>&& config,
+                  Layouter<F>* layouter) const override {
+    Fibonacci1Chip<F> fibonacci1_chip(std::move(config));
+
+    AssignedRow<F> first_row =
+        fibonacci1_chip.AssignFirstRow(layouter->Namespace("first row").get());
+
+    AssignedCell<F>& prev_b = first_row.b;
+    AssignedCell<F>& prev_c = first_row.c;
```

Comment:
Please remove `&`, otherwise `prev_c` points to the element that goes out of the scope

```suggestion
    AssignedCell<F> prev_b = first_row.b;
    AssignedCell<F> prev_c = first_row.c;
```